Sources/App/Controllers/FormatAPIController.swift

```diff
@@ -0,0 +1,222 @@
+        }
+        
+        let formatter = DateFormatter()
+        formatter.calendar = calendar
+        formatter.timeZone = timeZone
+        formatter.locale = locale
+        
+        switch request.format {
+            case .date(let date):
+                formatter.dateStyle = date.dateFormatterStyle
+                formatter.timeStyle = .none
+            case .time(let time):
+                formatter.dateStyle = .none
+                formatter.timeStyle = time.dateFormatterStyle
+            case .dateAndTime(let date, let time):
+                formatter.dateStyle = date.dateFormatterStyle
+                formatter.timeStyle = time.dateFormatterStyle
+            case .template(let string):
+                formatter.dateFormat = DateFormatter.dateFormat(fromTemplate: string, options: 0, locale: locale)
+            case .raw(let string):
+                formatter.dateFormat = string
+        }
+        
+        let date: Date
+        if let timestamp = request.timestamp {
+            date = Date(timeIntervalSince1970: timestamp)
+        } else {
+            date = Date()
+        }
+        
+        let formatted = formatter.string(from: date)
+        
+        return FormatResponse(id: request.id,
+                              calendar: calendarID,
+                              timeZone: timeZone.identifier,
+                              locale: locale.identifier,
+                              timestamp: date.timeIntervalSince1970,
+                              format: formatter.dateFormat,
+                              value: formatted)
+    }
+}
+
+struct FormatRequest: Content {
+    let id: String?
+    let calendar: String?
+    let timeZone: String?
+    let locale: String?
+    let timestamp: Double?
+    let format: Format
+    
+    enum Style: String, Codable {
+        case full
+        case long
+        case medium
+        case short
+        
+        var dateFormatterStyle: DateFormatter.Style {
+            switch self {
+                case .full: return .full
+                case .long: return .long
+                case .medium: return .medium
+                case .short: return .short
+            }
+        }
+    }
+    
+    enum Format: Codable {
+        
+        enum CodingKeys: CodingKey {
+            case date
+            case time
+            case template
+            case raw
+        }
+        
+        case date(Style)
+        case time(Style)
+        case dateAndTime(Style, Style)
+        case template(String)
+        case raw(String)
+        
+        init(from decoder: Decoder) throws {
+            let container = try decoder.container(keyedBy: FormatRequest.Format.CodingKeys.self)
+            let allKeys = container.allKeys
+            
+            if allKeys == [.date] {
+                self = .date(try container.decode(Style.self, forKey: .date))
+            } else if allKeys == [.date, .time] || allKeys == [.time, .date] {
```

Comment:
That was how I had originally written it, but I took it out because I figured the combinations were so minimal (only 6 possible variations, and never an array longer than 2 values) that it wasn't worth the extra CPU time to turn things into sets for "proper" matching.

I'm happy to put it back in if you're prefer it that way.